src/components/TextInput/TextInput.tsx

```diff
@@ -243,23 +251,35 @@ const TextInput = ({
           debouncedChange(e);
         }}
         multiLineIsResizable={multiLineIsResizable}
-        ref={inputRef}
+        ref={internalInputRef}
         {...inputProps}
       />
-      {onClear && (
-        <StyledIconContainer onClick={handleClear} {...iconContainerProps}>
+      {(onClear || clearable) && (
+        <StyledIconContainer
+          onClick={
+            onClear
+              ? (e: SyntheticEvent<Element, Event>) => handleClear(e)
+              : () => {
+                  if (internalInputRef?.current?.value) {
```

Comment:
We also want to check if value was provided through the props (i.e. controlled component) because we only want to clear the input if it's uncontrolled and value is undefined